src/main/java/org/acme/infinispanclient/service/HogwartsMagicCreator.java

```diff
@@ -0,0 +1,66 @@
+package org.acme.infinispanclient.service;
+
+import java.util.Random;
+import java.util.UUID;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.acme.infinispanclient.model.HPCharacter;
+import org.acme.infinispanclient.model.HPMagic;
+import org.acme.infinispanclient.model.HPSpell;
+import org.infinispan.client.hotrod.RemoteCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.quarkus.infinispan.client.runtime.Remote;
+import io.quarkus.scheduler.Scheduled;
+
+@ApplicationScoped
+public class HogwartsMagicCreator {
+   private static final Logger LOGGER = LoggerFactory.getLogger("HogwartsMagicCreator");
+
+   @Inject
+   @Remote(DataLoader.HP_CHARACTERS_NAME)
+   private RemoteCache<Integer, HPCharacter> characters;
+
+   @Inject
+   @Remote(DataLoader.HP_SPELLS_NAME)
+   private RemoteCache<Integer, HPSpell> spells;
+
+   @Inject
+   @Remote(DataLoader.HP_MAGIC_NAME)
+   private RemoteCache<String, HPMagic> magic;
+
+   private Random randomCharacters = new Random();
+   private Random randomSpells = new Random();
+
+   @Scheduled(every = "3s")
+   void executeMagic() {
+      LOGGER.info("... magic is happening ...");
+      Integer characterId = randomCharacters.nextInt(65);
```

Comment:
Would be nice if we didn't have to hard code the number of characters and spells.

Reply:
done